Added file on movedim (#5734)
Codecademy/docs: content/pytorch/concepts/tensor-operations/terms/movedim/movedim.md

---
Title: '.movedim()'
Description: 'A function used to reorder the dimensions of a tensor.'
Subjects:
  - 'AI'
  - 'Data Science'
Tags:
  - 'AI'
  - 'Arrays'
  - 'Data Structures'
  - 'Deep Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/computer-science'
---

**`.movedim()`** is used to move specific dimension of the input to a specified position. The other dimensions that are not explicitly mentioned remain in the original order.

## Syntax

```pseudo
torch.movedim(input, source, destination)
```

> - `input`: The input tensor.
> - `source`: Original position of the dimensions that need to move. It must be unique.
> - `destination`: Destination position for each of the original dimension. It must be unique.

Reviewer: ```suggestion
- `input`: The input tensor whose dimensions are to be rearranged.
- `source`: The dimensions to be moved. Can be a single integer or a tuple of integers.
- `destination`: The target positions for the dimensions specified in `source`. It should have the same length as `source`.
```

Reviewer: reframed for better clarity

## Example

The following example demonstrates the use of `.movedim()`:

```py
import torch

# Define a 1D tensor
a = torch.tensor([[1, 2, 3, -8]])

# Define a 2D tensor
b = torch.tensor([[1, 2, 3, -8],
                    [4, 3, 8, 0],
                    [-1, 7, 6, 3],
                    [5, 6, 9, 0]])

# Define a 3D tensor
c = torch.randn(2, 2, 3)

# Define a 4D tensor
d = torch.randn(2, 3, 2, 3)

# Move dimension 0 to dimension 1 for 1D tensor
a1 = torch.movedim(a, 0, 1)
print("One Dimensional tensor:")
print(a1)
print("\n")

# Move dimension 0 to dimension 1 for 2D tensor
b1 = torch.movedim(b, 0, 1)
print("Two Dimensional tensor:")
print(b1)
print("\n")

# Move dimension 0 to dimension 1 for 3D tensor
c1 = torch.movedim(c, 0, 1)
print("Three Dimensional tensor (Dim 1):")
print(c1)
print("\n")

# Move dimension 0 to dimension 2 for 3D tensor
c2 = torch.movedim(c, 0, 2)
print("Three Dimensional tensor (Dim 2):")
print(c2)
print("\n")

> # Move dimensions [0, 1] to positions [2, 3]

Reviewer: ```suggestion
# Move dimensions [0, 1] to positions [2, 3] for 4D tensor
```

d1 = torch.movedim(d, [0, 1], [2, 3])
print("Four Dimensional tensor:")
print(d1)
```

> 

Reviewer 2: ```suggestion
This example will generate the following output:

```

## Example

The following example demonstrates the use of `.movedim()`:

```py
import torch

# Define a 1D tensor
a = torch.tensor([[1, 2, 3, -8]])

# Define a 2D tensor
b = torch.tensor([[1, 2, 3, -8],
                    [4, 3, 8, 0],
                    [-1, 7, 6, 3],
                    [5, 6, 9, 0]])

# Define a 3D tensor
c = torch.randn(2, 2, 3)

# Define a 4D tensor
d = torch.randn(2, 3, 2, 3)

# Move dimension 0 to dimension 1 for 1D tensor
a1 = torch.movedim(a, 0, 1)
print("One Dimensional tensor:")
print(a1)
print("\n")

# Move dimension 0 to dimension 1 for 2D tensor
b1 = torch.movedim(b, 0, 1)
print("Two Dimensional tensor:")
print(b1)
print("\n")

# Move dimension 0 to dimension 1 for 3D tensor
c1 = torch.movedim(c, 0, 1)
print("Three Dimensional tensor (Dim 1):")
print(c1)
print("\n")

# Move dimension 0 to dimension 2 for 3D tensor
c2 = torch.movedim(c, 0, 2)
print("Three Dimensional tensor (Dim 2):")
print(c2)
print("\n")

# Move dimensions [0, 1] to positions [2, 3] for 4D tensor
d1 = torch.movedim(d, [0, 1], [2, 3])
print("Four Dimensional tensor:")
print(d1)
```

> This example will generate the following output:

Reviewer 2: ```suggestion

This example will generate the following output:
```
